tests/CMakeLists.txt

```diff
@@ -57,6 +57,12 @@ target_compile_options(gsl_tests_config INTERFACE
         -Wshadow
         -Wsign-conversion
     >
+    $<$<CXX_COMPILER_ID:Clang>:
```

Comment:
Can you please clarify why those are needed - anything different in clang-cl that we didn't need in clang? or something added in latest version of clang?

Reply:
_A case of "works on my machine"-reversed._
Neither, those were needed to compile with `/Weverything`.

When compiling with CMake in Visual Studio these warnings where show. Not on AppVeyor.
They should not unless `/Weverything` is active, but that flag is not set, (`-v` on ninja)
... Deleting the CMakeCache solved it. Bug in VS's CMake?

Now I have to check if any other change was unnecessary ...
On the bright-side, disabling 6 warnings is enough to compile with every warning in clang enabled.

Reply:
> ... Deleting the CMakeCache solved it. Bug in VS's CMake?

Possible, do you think you have a repro, or repro steps? You can use "clone GSL" as a repro step. I can send it to the CMake team.

Reply:
> On the bright-side, improvements have been made, and disabling 7 warnings is enough to compile with /WX and every warning in clang enabled.

Great! As for the unnecessary disabling, we should re-enable them if the don't fire. To be sure we do not  miss warnings in the future.

Reply:
> repro, or repro steps?

I'll give it a try, when I have some time.
> As for the unnecessary disabling, we should re-enable them if they don't fire. To be sure we do not miss warnings in the future.

Done, pushed the warning-level with clang-cl to -Weverything.
Nothing special needed, just adding a `#if defined(__GNUC__)` in a few places to solve `-Wundef`.